packages/core/useElementBySelector/index.ts

```diff
@@ -0,0 +1,18 @@
+import type { MaybeRefOrGetter } from 'vue'
+import { tryOnMounted } from '@vueuse/shared'
+import { ref as deepRef, toValue, watch } from 'vue'
+
+export function useElementBySelector(selector: MaybeRefOrGetter<string>) {
+  const el = deepRef<Element | null>()
+  const select = (): Element | null => el.value = document.querySelector(toValue(selector))
```

Comment:
lets do this:
```suggestion
  const select = (newSelector): Element | null => el.value = document.querySelector(newSelector)
```
this ensures the right value for the current watch cycle

Reply:
Please checkout the guidelines for globals like document: https://vueuse.org/guidelines.html#configurable-globals

Reply:
Updated 🙏🏻